nuitka/tree/ReformulationFunctionStatements.py

```diff
@@ -643,9 +735,15 @@ def extractArgAnnotation(arg):
         )
 
     if keys:
-        return makeDictCreationOrConstant2(
-            keys=keys, values=values, source_ref=source_ref
-        )
+        # On 3.14+, annotations are deferred by default.
+        if python_version >= 0x3E0:
+            return makeDeferredAnnotateFunction(
```

Comment:
Are you saying, that all this is doing, is to create a function that returns the dictionary, just later? I thought the actual code of the annotations is to be executed delayed.

```
def f():
   x : something()

f()
f.__annotations__
```

I was assuming this will not crash when calling f().

Reply:
On second thought, that might be already correct for functions if "keys" and "values" are the ones from the function definition, not sure now, but it definitely doesn't do classes yet. And for empty annotations, I think it ought not to create a function, but instead end up passing a value. Or we check if that functions becomes a constant returner, in which case we ought to just pass its value or something, but that is optimization, which should for now maybe only result in TODOs. But a new function object per function, we don't want to hurt performance that much if we can help it in any way.